step-44: Change the FE space used for p,J

jppelteret opened the pull request:
Fixes #11612

kronbichler reviewed (approved):
I agree with this change, this makes a lot of sense (the condition numbers can be hilarious with FE_DGPMonomial on bigger meshes).

kronbichler commented on doc/news/changes/minor/20220426Jean-PaulPelteret:
```diff
@@ -0,0 +1,6 @@
+Improved: The step-44 tutorial now uses FE_DGP elements to discretise the
```
I am not sure if we strive for American English consistently, but let me suggest it anyway :wink:
```suggestion
Improved: The step-44 tutorial now uses FE_DGP elements to discretize the
```

jppelteret replied:
Thanks, I've now fixed this.

bangerth commented:
I agree with the change to step-44 as well. For the tests, my preference would have been to duplicate the existing tests, rather than change them. There is value in ensuring that both approaches continue to work, regardless of whether they are the "better" or "worse" way to approach things.

jppelteret commented:
Hi @bangerth,

> For the tests, my preference would have been to duplicate the existing tests, rather than change them.

Fair point. I've duplicated just the test `physics/step-44.cc` to `physics/step-44-dgp_monomial.cc` which now gives coverage to both implementations. Does that work for you?

jppelteret commented:
Hi @kronbichler,

Thanks for the review!

> the condition numbers can be hilarious with FE_DGPMonomial on bigger meshes

I didn't know that it could get that bad. I swear that when we implemented the tutorial we started off with `FE_DGP` but didn't get any results, and then we switched to `FE_DGPMonomial` and suddenly we got a solution. I don't know why that would be, but clearly that's no longer the case.

jppelteret commented:
/rebuild
